File: Creational/Builder/Java/ConcreteBuilder1.java

public class ConcreteBuilder1 extends AbstractBuilder{

	@Override
	public void reset() {
		this.abstractProduct = new ConcreteProduct1();
		
	}

	@Override
	public void buildStepA() {
		this.abstractProduct.stepA1();
	}

	@Override
	public void buildStepB() {
		this.abstractProduct.stepB1();
		
	}

	@Override
	public void buildStepC() {
		this.abstractProduct.stepC1();
	}

}

File: Creational/Builder/Kotlin/ConcreteBuilder1.kt
class ConcreteBuilder1: AbstractBuilder() {

    override fun reset() {
        this.abstractProduct = ConcreteProduct1()
    }

    override fun buildStepA() {
        this.abstractProduct.stepA1()
    }

    override fun buildStepB() {
        this.abstractProduct.stepB1()
    }

    override fun buildStepC() {
        this.abstractProduct.stepC1()
    }

}